src/figuresClasses.ts

```diff
@@ -1,11 +1,92 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public firstSide: number,
+    public secondSide: number,
+    public thirdSide: number,
+  ) {
+    const biggestSide: number = Math.max(firstSide, secondSide, thirdSide);
+    const otherSidesSum: number = [firstSide, secondSide, thirdSide]
+      .filter((side) => side !== biggestSide)
+      .reduce((acc, side) => acc + side);
+
+    if (firstSide <= 0 || secondSide <= 0 || thirdSide <= 0) {
+      throw new Error('all sides must be positive');
+    }
+
+    if (biggestSide >= otherSidesSum) {
+      throw new Error('this sides cannot form a triangle');
+    }
+  }
+
+  getArea(): number {
```

Comment:
[CODE STYLE] - The `getArea` method should not accept any parameters as all necessary values are already available as object fields. Remove the destructuring assignment in line 34 since you can directly access `firstSide`, `secondSide`, and `thirdSide` on `this`.